src/Amazon.AspNetCore.Identity.Cognito/CognitoUserManager.cs

```diff
@@ -180,7 +178,7 @@ private async Task PopulateTokens(TUser user, string claimType, string claimValu
         /// <returns>The <see cref="Task"/> that represents the asynchronous operation, containing the AuthFlowResponse object
         /// if the specified <paramref name="password" /> matches the one store for the <paramref name="user"/>,
         /// otherwise null.</returns>
-        public virtual Task<AuthFlowResponse> CheckPasswordAsync(TUser user, string password)
+        public new virtual Task<AuthFlowResponse> CheckPasswordAsync(TUser user, string password)
```

Comment:
I assume you added the `new` to address a warning. This seems bad we are using `new virtual` which could cause problems with in some scenarios the base class that returns back `Task<bool>` is being called. I'm not sure the base class version will even work for Cognito. I don't want to expand your work with an unrelated change but doing a major version bump might be the opportunity to fix this. Can you add a story to the input queue replace the `new virtual` with a way for us to `override` the base class?

Reply:
I have created a backlog item for us to address this.